static/app/views/replays/detail/breadcrumbs/useBreadcrumbFilters.tsx

```diff
@@ -0,0 +1,89 @@
+import {useCallback, useMemo} from 'react';
+import uniq from 'lodash/uniq';
+import upperFirst from 'lodash/upperFirst';
+
+import {decodeList, decodeScalar} from 'sentry/utils/queryString';
+import getFrameDetails from 'sentry/utils/replays/getFrameDetails';
+import useFiltersInLocationQuery from 'sentry/utils/replays/hooks/useFiltersInLocationQuery';
+import {getFrameOpOrCategory, ReplayFrame} from 'sentry/utils/replays/types';
+import {filterItems} from 'sentry/views/replays/detail/utils';
+
+export type FilterFields = {
+  f_b_search: string;
+  f_b_type: string[];
+};
+
+type Options = {
+  frames: ReplayFrame[];
+};
+
+type Return = {
+  getBreadcrumbTypes: () => {label: string; value: string}[];
+  items: ReplayFrame[];
+  searchTerm: string;
+  setSearchTerm: (searchTerm: string) => void;
+  setType: (type: string[]) => void;
+  type: string[];
+};
+
+const FILTERS = {
+  type: (item: ReplayFrame, type: string[]) =>
+    type.length === 0 || type.includes(getFrameOpOrCategory(item)),
+  searchTerm: (item: ReplayFrame, searchTerm: string) =>
+    JSON.stringify(item).toLowerCase().includes(searchTerm),
+};
+
+function useBreadcrumbFilters({frames}: Options): Return {
+  const {setFilter, query} = useFiltersInLocationQuery<FilterFields>();
+
+  const type = useMemo(() => decodeList(query.f_b_type), [query.f_b_type]);
+  const searchTerm = decodeScalar(query.f_b_search, '').toLowerCase();
+
+  const items = useMemo(
+    () =>
+      filterItems({
+        items: frames,
+        filterFns: FILTERS,
+        filterVals: {type, searchTerm},
+      }),
+    [frames, type, searchTerm]
+  );
+
+  const TYPE_TO_LABEL: Record<string, string> = useMemo(() => ({}), []);
```

Comment:
This looks funny. 

Looks like you're trying to keep the same object ref between renders, and rely on the object being mutated to insert all the fields. 

It's cool, but the useMemo part isn't needed IMO because we're not saving any computation behind it, the forEach is the compute part and it is running each time across all unfiltered frames. 

useNetworkFilter should be doing the same thing for its types, that might be a good example, I can check tmrw though.

Reply:
You know what, this is more interesting than I thought last night.
I bet we could be doing better in the existing examples to memoize the data (right now we're looping over all frames whenever the callback is called, which could be a few times).
So maybe we will merge this because it works, then return later to profile.